astropy/cosmology/io/html.py

```diff
@@ -0,0 +1,140 @@
+import astropy.cosmology.units as cu
+import astropy.units as u
+from astropy.cosmology.connect import readwrite_registry
+from astropy.cosmology.core import Cosmology
+from astropy.cosmology.parameter import Parameter
+from astropy.table import QTable
+
+from .table import from_table, to_table
+
+# format look-up for conversion, {original_name:new_name}
+format_table = {"H0": "$$H_{0}$$",
+                "Om0": "$$\\Omega_{m,0}$$",
+                "Ode0": "$$\\Omega_{\\Lambda,0}$$",
+                "Tcmb0": "$$T_{0}$$",
+                "Neff": "$$N_{eff}$$",
+                "m_nu": "$$m_{nu}$$",
+                "Ob0": "$$\\Omega_{b,0}$$",
+                "w0": "$$w_{0}$$",
+                "wa": "$$w_{a}$$",
+                "wz": "$$w_{z}$$",
+                "wp": "$$w_{p}$$",
+                "zp": "$$z_{p}$$"}
+
+
+def read_html_table(filename, index=None, *, move_to_meta=False, cosmology=None, **kwargs):
+    """Read a `~astropy.cosmology.Cosmology` from an HTML file.
+
+    Parameters
+    ----------
+    filename : path-like or file-like
+        From where to read the Cosmology.
+    index : int, str, or None, optional
+        Needed to select the row in tables with multiple rows. ``index`` can be
+        an integer for the row number or, if the table is indexed by a column,
+        the value of that column. If the table is not indexed and ``index``
+        is a string, the "name" column is used as the indexing column.
+
+    move_to_meta : bool (optional, keyword-only)
+        Whether to move keyword arguments that are not in the Cosmology class'
+        signature to the Cosmology's metadata. This will only be applied if the
+        Cosmology does NOT have a keyword-only argument (e.g. ``**kwargs``).
+        Arguments moved to the metadata will be merged with existing metadata,
+        preferring specified metadata in the case of a merge conflict
+        (e.g. for ``Cosmology(meta={'key':10}, key=42)``, the ``Cosmology.meta``
+        will be ``{'key': 10}``).
+
+    cosmology : str, `~astropy.cosmology.Cosmology` class, or None (optional, keyword-only)
+        The cosmology class (or string name thereof) to use when constructing
+        the cosmology instance. The class also provides default parameter values,
+        filling in any non-mandatory arguments missing in 'table'.
+
+    **kwargs
+        Passed to :attr:`astropy.table.QTable.read`
+
+    Returns
+    -------
+    `~astropy.cosmology.Cosmology` subclass instance
+    """
+    kwargs["format"] = "ascii.html"
+    with u.add_enabled_units(cu):
+        table = QTable.read(filename, **kwargs)
+
+    table_columns = set(table.columns)
+
+    for old_name, new_name in format_table.items():
+        if new_name in table_columns:
+            table.rename_column(new_name, old_name)
+
+    # build cosmology from table
+    return from_table(table, index=index, move_to_meta=move_to_meta, cosmology=cosmology)
+
+
+def write_html_table(

```

Comment:
I think for this function it would be good to have an ``Examples`` or ``Notes`` Section (https://numpydoc.readthedocs.io/en/latest/format.html#examples). In particular I think it's necessary to explain that the HTML code requires
```
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script type="text/javascript" id="MathJax-script" async
  src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml.js">
</script>
```

to be in the file header to display properly. Many websites with embedded math will already have this, but for those that don't it's a useful reference.